indra/llcommon/lua_function.cpp

```diff
@@ -793,36 +821,25 @@ LuaState::~LuaState()
         LL_DEBUGS("Lua") << LLCoros::getName() << ": Registry.atexit is a table with "
                          << len << " entries" << LL_ENDL;
 
-        // Push debug.traceback() onto the stack as lua_pcall()'s error
-        // handler function. On error, lua_pcall() calls the specified error
-        // handler function with the original error message; the message
-        // returned by the error handler is then returned by lua_pcall().
-        // Luau's debug.traceback() is called with a message to prepend to the
-        // returned traceback string. Almost as if they'd been designed to
-        // work together...
-        lua_getglobal(mState, "debug");
-        lua_getfield(mState, -1, "traceback");
-        // ditch "debug"
-        lua_remove(mState, -2);
-        // stack now contains atexit, debug.traceback()
-
+        // TODO: 'debug' global shouldn't be overwritten and should be accessible at this stage
+        bool debug_traceback = push_debug_traceback();
+        // if debug_traceback is true, stack now contains atexit, /debug.traceback()/
+        // otherwise just atexit
         for (int i(len); i >= 1; --i)
         {
             lua_pushinteger(mState, i);
-            // stack contains Registry.atexit, debug.traceback(), i
-            lua_gettable(mState, -3);
-            // stack contains Registry.atexit, debug.traceback(), atexit[i]
+            // stack contains Registry.atexit, /debug.traceback()/, i
+            lua_gettable(mState, debug_traceback ? -3 : -2);
```

Comment:
No - up around line 815, right after `if (lua_istable(mState, -1)) {`, capture `int atexit = lua_gettop(mState);`

Then you can use `atexit` here.